common/src/main/java/io/netty/util/ResourceLeakDetectorFactory.java

```diff
@@ -101,20 +112,25 @@ private Constructor customClassConstructor() {
         }
 
         @Override
-        public <T> ResourceLeakDetector<T> newResourceLeakDetector(final Class<T> resource) {
+        public <T> ResourceLeakDetector<T> newResourceLeakDetector(
+                Class<T> resource, int samplingInterval, long maxActive) {
             try {
                 if (customClassConstructor != null) {
+                    @SuppressWarnings("unchecked")
                     ResourceLeakDetector<T> leakDetector =
-                            (ResourceLeakDetector<T>) customClassConstructor.newInstance(resource);
-                    logger.debug("Loaded custom ResourceLeakDetector: {}", customLeakDetector);
+                            (ResourceLeakDetector<T>) customClassConstructor.newInstance(
+                                    resource, samplingInterval, maxActive);
+                    logger.debug("Loaded custom ResourceLeakDetector: {}",
+                            customClassConstructor.getDeclaringClass().getName());
                     return leakDetector;
                 }
             } catch (Throwable t) {
                 logger.error("Could not load custom resource leak detector provided: {} with the given resource: {}",
-                        customLeakDetector, resource, t);
+                        customClassConstructor.getDeclaringClass().getName(), resource, t);
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) NullPointerException might be thrown as 'customClassConstructor' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS2259)